sdk/keyvault/keyvault-keys/src/cryptographyClient.ts

```diff
@@ -628,9 +608,127 @@ export interface EncryptOptions extends RequestOptions {
 }
 
 /**
- * Options for the decrypt call to the CryptographyClient
+ * @internal
+ * @ignore
+ * Encodes a length of a packet in DER format
+*/
+function encodeLength(length: number): Uint8Array {
+  if (length <= 127) {
+    return Uint8Array.of(length);
+  } else if (length < 256) {
+    return Uint8Array.of(0x81, length);
+  } else if (length < 65536) {
+    return Uint8Array.of(0x82, length >> 8, length & 0xff);
+  } else {
+    throw new Error("Unsupported length to encode");
+  }
+}
+
+/**
+ * @internal
+ * @ignore
+ * Encodes a buffer for DER, as sets the id to the given id
  */
+function encodeBuffer(buffer: Uint8Array, bufferId: number): Uint8Array {
+  if (buffer.length == 0) {
+      return buffer;
+  }
+
+  let result = new Uint8Array(buffer);
+  
+  // If the high bit is set, prepend a 0
+  if ((result[0] & 0x80) === 0x80) {
+      let array = new Uint8Array(result.length + 1);
+      array[0] = 0;
+      array.set(result, 1);
+      result = array;
+  }
+
+  // Prepend the DER header for this buffer
+  let encodedLength = encodeLength(result.length);
 
+  let totalLength = 1 + encodedLength.length + result.length;
+
+  let outputBuffer = new Uint8Array(totalLength);
+  outputBuffer[0] = bufferId; 
+  outputBuffer.set(encodedLength, 1);
+  outputBuffer.set(result, 1 + encodedLength.length);
+
+  return outputBuffer;
+}
+
+/**
+ * @internal
+ * @ignore
+ * Encode a JWK to PEM format. To do so, it internally repackages the JWK as a DER
+ * that is then encoded as a PEM.
+ */
+export function convertJWKtoPEM(key: JsonWebKey): string {
+  if (!key.n || !key.e) {
+    throw new Error("Unsupported key format for local operations");
+  }
+  let encoded_n = encodeBuffer(key.n, 0x2); // INTEGER
+  let encoded_e = encodeBuffer(key.e, 0x2); // INTEGER
+
+  let encoded_ne = new Uint8Array(encoded_n.length + encoded_e.length);
+  encoded_ne.set(encoded_n, 0);
+  encoded_ne.set(encoded_e, encoded_n.length);
+
+  let full_encoded = encodeBuffer(encoded_ne, 0x30); //SEQUENCE
+
+  let buffer = Buffer.from(full_encoded).toString("base64");
+
+  let beginBanner = "-----BEGIN RSA PUBLIC KEY-----\n";
```

Comment:
It's possible that this would contain other things besides an RSA key, for instance an EC key, and in the near future an AES key. I'm guessing that this is all we need for now because we only support RSA local crypto?

Reply:
Exactly. Right now, it's made-to-order for the keys we support. As that set grows, we can update the converter to match.